derotation/simulate/basic_rotator.py

```diff
@@ -0,0 +1,112 @@
+import copy
+
+import numpy as np
+from scipy.ndimage import affine_transform
+
+
+class Rotator:
+    def __init__(self, angles: np.ndarray, image_stack: np.ndarray) -> None:
+        """Initializes the Rotator object.
+
+        Parameters
+        ----------
+        angles : np.ndarray
+            The rotation angles by line per frame
+        image_stack : np.ndarray
+            The image stack to be rotated
+
+        Raises
+        ------
+        AssertionError
+            If the number of angles is not equal to the number of lines
+            per frame multiplied by the number of frames
+        """
+        #  there should be one angle per line pe frame
+        assert len(angles) == image_stack.shape[0] * image_stack.shape[1], (
+            f"Number of angles ({len(angles)}) should be equal to the number "
+            + "of lines per frame multiplied by the number of frames "
+            + f"({image_stack.shape[0] * image_stack.shape[1]})"
+        )
+
+        self.angles = angles
+        self.image_stack = image_stack
+        self.num_lines_per_frame = image_stack.shape[1]
+
+    def rotate_by_line(self) -> np.ndarray:
+        """Rotates the image stack line by line, using the rotation angles
+        provided.
```

Comment:
Thank you Joe, I think that I will add more docstrings and also include animations and explanations, at the top level. I could write it in the readme but it will explode in content. It might be time to start to write some proper documentation 😊

Reply:
issue -> #29